### PR TITLE
You can no longer sentience potion humanoid simplemobs

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

adds a new sentience type for humanoids that isnt able to be sentienced, adds it to humanoid mobs, standardizes some of the use of the human mob biotypes and removes dark wizards from the gold slime pool, since they are no longer able to be sentienced

## Why It's Good For The Game

you should not be able to walk up to a syndicate stormtrooper and click on him with a potion to make him crew aligned and your shotgun slave

## Changelog
:cl:
balance: You can no longer sentience potion humanoid simplemobs
/:cl:

### CODE REVIEW

**Comment by Qustinnus** [comment]
noooo dark wizards!!!!!

**Comment by Livrah** [comment]
According to playlist, Mayhem in the Bottle will be next.

**Comment by D4C-420** [comment]
> you should not be able to walk up to a syndicate stormtrooper and click on him with a potion to make him crew aligned and your shotgun slave

if your going through the effort to grab syndicate stormtroopers you deserve them. Why does this have to effect dark wizards too? As they are their balance is fine.  this smells of ided.

**Comment by Fikou** [comment]
> if your going through the effort to grab syndicate stormtroopers you deserve them. 

why, its a literal syndicate commando, its supposed to be a goddamn human npc replacement, you cannot sentience humans and this is just a way to create npc mobs without coding more human stuff, you are possessing basically a nukie

> As they are their balance is fine

what fine balance is there to speak of its just a human wizard shooting bolts from a lavaland ruin that was randomly put in gold slime pool

> this smells of ided.

i have not died to any of these mobs while they were sentient :)

**Comment by Fikou** [comment]
> According to playlist, Mayhem in the Bottle will be next.

sorry the world does not revolve around you i have not watched whatever video you made about this

**Comment by D4C-420** [comment]
> why, its a literal syndicate commando, its supposed to be a goddamn human npc replacement, you cannot sentience humans and this is just a way to create npc mobs without coding more human stuff, you are possessing basically a nukie

these things are abled to be killed just fine by the crew. You can't really compare it to a nukie. 

> what fine balance is there to speak of its just a human wizard shooting bolts from a lavaland ruin that was randomly put in gold slime pool

The gold slime pool is filled with random stuff already. The dark wizard is far from the worst offender on that list. Really the dark wizards are not that strong.

**Comment by moocowswag** [comment]
So from what i can tell.

-The mobs effected are limited in supply/not too strong or overbearing. So its not an emergency balance change.

-They do not have the same capabilities to manipulate their environment as an actual humanoid.

-Sentience potions not being allowed to sentience humanoids is to prevent mass humanified monkeys from being sentienced so nobody can easily repopulate the station with perfect dextrous/asimov protected crew

-The pr goes against consistency and creates a snowflake simple mob subtype that are considered humanoids just because of their sprite when functionaly they couldnt be farther from it.

**Comment by Qustinnus** [comment]
its not snowflake. thats what biotypes and sentience types are for.

**Comment by moocowswag** [comment]
> its not snowflake. thats what biotypes and sentience types are for.

I mean in game, it looks like a simple mob it, it acts like a simple mob, it even explodes into gore like a simple mob, but its somehow special because it's sprite is shaped like a humanoid.

**Comment by Qustinnus** [comment]
what kind of stupid logic is that? Simple mob just implies it doesn't require advanced handling like organs because it's a mob almost only used for combat.

Do you understand what a sentience potion is? It makes things sentient. Humans already are sentient. Whether it's literaly the human subtype or not doesn't matter. If we had the processing power those simplemobs would also be that subtype.

**Comment by moocowswag** [comment]
>Humans already are sentient.

Considering what the syndicate are able to do to people's minds and how that is implied through varous game mechanics I very much doubt the guys clad in armor mindlessly watching over a crashed ship are anywhere near sentient.

**Comment by Qustinnus** [comment]
I, the coder, say they are. What now liberal?

**Comment by ATH1909** [comment]
> I, the coder, say they are. What now liberal?

I, another coder, say that they are indeed mindless, brainwashed drones. What now conservative?

**Comment by Qustinnus** [comment]
image

**Comment by ATH1909** [comment]
In general, I really don't see a balance reason for this PR, and I think that the rule of cool outweighs the minor fluff/lore reason for this PR, especially since we can just change the fluff/lore to fit our actual game mechanics to get the best of both worlds.

**Comment by ATH1909** [comment]
image

Don't throw stones in glass houses, Floyd.

**Comment by Fikou** [comment]
> In general, I really don't see a balance reason for this PR, and I think that the rule of cool outweighs the minor fluff/lore reason for this PR, especially since we can just change the fluff/lore to fit our actual game mechanics to get the best of both worlds.

wheres the rule of cool theres no skeletons shredding guitars its just people giving sentience to syndie stormtroopers (mobs which already have sentience from the ingame world perspective) to make them bar rp

**Comment by ATH1909** [comment]
> > In general, I really don't see a balance reason for this PR, and I think that the rule of cool outweighs the minor fluff/lore reason for this PR, especially since we can just change the fluff/lore to fit our actual game mechanics to get the best of both worlds.
> 
> wheres the rule of cool theres no skeletons shredding guitars its just people giving sentience to syndie stormtroopers (mobs which already have sentience from the ingame world perspective) to make them bar rp

https://youtu.be/CWr5fgxRsZU

**Comment by Qustinnus** [comment]
dont need maint rank to make sense

**Comment by ATH1909** [comment]
> dont need maint rank to make sense

the same would thus apply to me, then, no?

**Comment by Qustinnus** [comment]
yeah but you actively prove you don't in almost every post you make.

**Comment by ATH1909** [comment]
> yeah but you actively prove you don't in almost every post you make.

😢

**Comment by LemonInTheDark** [comment]
A: Christ almighty you went so far off topic I can't see the sun anymore
B: Holy fucking shit stop arguing that simple mobs are anything more then an optimization jesus christ they are not a "feature"

**Comment by ishitbyabullet** [comment]
This kind of shit is like a joke argument that could be twisted to justify why you should be able to use the sentience potion on assistants. You're obviously not supposed to be able to fucking convert human enemies into sentient because humans should already be sentient as according to common fucking sense.

Also, D4C-420, are you a sadist? Yes, it is possible to take on simplemobs however a syndicate stormtrooper with a shotgun is basically an infinite ammo shotgunner with a lot of health. I've seen a sentient syndicate shotgunner literally **wipe out an entire away mission exploration team** because they weren't loaded with gamer loot to counter the sentient simplemob.

**Comment by ATH1909** [comment]
> I've seen a sentient syndicate shotgunner literally **wipe out an entire away mission exploration team** because they weren't loaded with gamer loot to counter the sentient simplemob.

nice

**Comment by D4C-420** [comment]
> Also, D4C-420, are you a sadist? Yes, it is possible to take on simplemobs however a syndicate stormtrooper with a shotgun is basically an infinite ammo shotgunner with a lot of health. I've seen a sentient syndicate shotgunner literally wipe out an entire away mission exploration team because they weren't loaded with gamer loot to counter the sentient simplemob.

No I’m a decent miner and saying that one sentient shot gunner wiping out a whole exploration team is justification for not allowing sentience on them is not much a good argument. Did the team have guns? What was their armor. How hurt were they before hand. Did they suffer from any kind of slow down? What was their healing like? How about environment?  Where they forced into close combat? Picked off one by one? What I’m saying here is that most of the time beating up a simple mob is a matter of being smart enough to take advantage of its limited capabilities even if it’s sent it. An infinite ammo shot gunner is for all intents and purposes just a man with a lot of ammo and instead of damage reduction and healing it has extra health. And for the same reason you see fluke ops happen often you can beat even the more dangerous simple mobs. The nuke op makes a dumb mistake like getting close to the guy with a bat and then getting stun locked to death after being beat into a wall. It’s a matter of taking advantage of your capabilities and your enemies weakness. And no simple mob killing “gamer loot” would not be more effective on them than a human. Pka’s deal damage bases off of whether or not your fighting in low pressure and last I checked non of these mobs could be marked and detonated upon by a KC. Also if this issue is such a big issue why does it have to also effect dark wizards? That just seems like limiting player choice for the sake of consistency rather than any meaningful gameplay changes.
 
Tldr: non-lava land simple mobs even sentient ones can still be beaten by fighting smarter and there is no reason for removing the dark wizard from the gold core pool.

File: code/modules/ruins/spaceruin_code/forgottenship.dm [hunk 1]
```diff
@@ -172,7 +172,6 @@ GLOBAL_VAR_INIT(fscpassword, generate_password())
 /mob/living/simple_animal/hostile/nanotrasen/elite
 	name = "Nanotrasen Elite Assault Officer"
 	desc = "Pray for your life, syndicate. Run while you can."
-	icon = 'icons/mob/simple_human.dmi'
```
Comment (Mothblocks) [inline code/modules/ruins/spaceruin_code/forgottenship.dm:175]: Have you confirmed this isn't necessary?

File: code/__DEFINES/mobs.dm [hunk 2]
```diff
@@ -229,7 +229,7 @@
 //Sentience types, to prevent things like sentience potions from giving bosses sentience
 #define SENTIENCE_ORGANIC 1
 #define SENTIENCE_ARTIFICIAL 2
-// #define SENTIENCE_OTHER 3 unused
+#define SENTIENCE_HUMANOID 3
```
Comment (Mothblocks) [inline code/__DEFINES/mobs.dm:232]: Was this checked somewhere before hand but with `3` directly? Why is your code only updating flags?

File: code/modules/ruins/spaceruin_code/forgottenship.dm (hunk 1 above)
Comment (Fikou) [reply code/modules/ruins/spaceruin_code/forgottenship.dm:175]: yes

File: code/__DEFINES/mobs.dm (hunk 2 above)
Comment (Fikou) [reply code/__DEFINES/mobs.dm:232]: the code mostly checks if the potion's sentience type is the same as the mob's (i guess to make it so admins can edit the needed sentience?), and its not set anywhere to 3 from a basic check

Comment (Mothblocks) [reply code/__DEFINES/mobs.dm:232]: Ah, ok.

**Comment by NanoCats** [comment]
I disagree with this as it reduces the capacity for amusing shenanigans but am ultimately powerless.
